### PR TITLE
Feat/format playtime in hours

### PR DESCRIPTION
**Pull request by zamitto**
- Changes how the playtime is formated to show in minutes (until 120 minutes) and in hours
- In the previous version of this PR I had increased the sleep time of process-watcher loop to 5 seconds. But we can't right now increase it too much as the game-details use this event to update the isGamePlaying status.

image

image

### CODE REVIEW

File: src/renderer/src/pages/game-details/hero-panel.tsx [hunk 1]
```diff
@@ -47,27 +47,29 @@ export function HeroPanel({
...
-      return () => {
-        clearInterval(interval);
-      };
-    }
-  }, [game?.lastTimePlayed, updateLastTimePlayed]);
+      if (minutes < 120) {
+        return minutes.toFixed(0) + " " + t("minutes")
+      }
+
+      const numberFormat = new Intl.NumberFormat(i18n.language, { maximumFractionDigits: 1 })
```
Comment (hydralauncher) [inline src/renderer/src/pages/game-details/hero-panel.tsx:68]: You should create the `NumberFormat` instance outside of the component so it doesn't have to instantiated every single time

File: src/renderer/src/pages/game-details/hero-panel.tsx [hunk 2]
```diff
@@ -47,27 +47,29 @@ export function HeroPanel({
...
+
+      const hours = minutes / 60
+      return numberFormat.format(hours)  + " " + t("hours")
```
Comment (hydralauncher) [inline src/renderer/src/pages/game-details/hero-panel.tsx:71]: Please use i18n interpolation instead:
```ts
t("playtime", { hours: minutes / 60 })
```

And on the translation file:
```json
{
"playtime": "{{hours}} hours"
}
```

This way you can handle multiple languages better

File: src/renderer/src/pages/game-details/hero-panel.tsx [hunk 3]
```diff
@@ -47,27 +47,29 @@ export function HeroPanel({
...
+  const formatPlayTime = (milliseconds: number) => {
+      const seconds = milliseconds / 1000
+      const minutes = seconds / 60
 
-      return () => {
-        clearInterval(interval);
-      };
-    }
-  }, [game?.lastTimePlayed, updateLastTimePlayed]);
+      if (minutes < 120) {
```
Comment (hydralauncher) [inline src/renderer/src/pages/game-details/hero-panel.tsx:64]: `120` is a magic number here. Please assign it to a constant outside of the function

File: src/renderer/src/pages/game-details/hero-panel.tsx [hunk 4]
```diff
@@ -47,27 +47,29 @@ export function HeroPanel({
...
+        return minutes.toFixed(0) + " " + t("minutes")
```
Comment (hydralauncher) [inline src/renderer/src/pages/game-details/hero-panel.tsx:65]: Consider using the same logic as I have mentioned for the hours

**Review by hydralauncher — CHANGES_REQUESTED** [review]

File: src/renderer/src/pages/game-details/hero-panel.tsx (hunk 1 above)
Comment (zamitto) [reply src/renderer/src/pages/game-details/hero-panel.tsx]: Not sure what's the best way to solve this as I don't know much of React, but I used a `useMemo` hook and put the `NumberFormat` inside it, what do you think?

**Review by hydralauncher — APPROVED** [review]
LGTM

**Pull request merged**